src/MemoryRouter.tsx

```diff
@@ -13,9 +13,13 @@ function getRouteAsPath(pathname: string, query: ParsedUrlQuery) {
 
   // Replace slugs, and remove them from the `query`
   let asPath = pathname.replace(/\[(.+?)]/g, ($0, slug: keyof ParsedUrlQuery) => {
+    if (String(slug).startsWith("...")) slug = String(slug).replace("...", "")
```

Comment:
Would it be hard here to also support "optional catch all" syntax? Eg `[[...slug]]`?